smac/facade/smac_facade.py

```diff
@@ -207,44 +253,64 @@ def __init__(self,
 
         # initial EPM
         types, bounds = get_types(scenario.cs, scenario.feature_array)
+        model_def_kwargs = {
+            'types':types,
+            'bounds':bounds,
+            'instance_features':scenario.feature_array,
+            'seed':rng.randint(MAXINT),
+            'pca_components':scenario.PCA_DIM,
+            'log_y':scenario.transform_y in ["LOG", "LOGS"],
+            'num_trees':scenario.rf_num_trees,
+            'do_bootstrapping':scenario.rf_do_bootstrapping,
+            'ratio_features':scenario.rf_ratio_features,
+            'min_samples_split':scenario.rf_min_samples_split,
+            'min_samples_leaf':scenario.rf_min_samples_leaf,
+            'max_depth':scenario.rf_max_depth
+            }
+        model_def_kwargs.update(model_kwargs)
         if model is None:
-            model = RandomForestWithInstances(types=types,
-                                              bounds=bounds,
-                                              instance_features=scenario.feature_array,
-                                              seed=rng.randint(MAXINT),
-                                              pca_components=scenario.PCA_DIM,
-                                              log_y=scenario.transform_y in ["LOG", "LOGS"],
-                                              num_trees=scenario.rf_num_trees,
-                                              do_bootstrapping=scenario.rf_do_bootstrapping,
-                                              ratio_features=scenario.rf_ratio_features,
-                                              min_samples_split=scenario.rf_min_samples_split,
-                                              min_samples_leaf=scenario.rf_min_samples_leaf,
-                                              max_depth=scenario.rf_max_depth)
+            model = RandomForestWithInstances(**model_def_kwargs)
+        elif inspect.isclass(model):
+            model = model(**model_def_kwargs)
```

Comment:
I'm wondering what happens if the model does not accept the number of trees as a hyperparameter? Maybe we need to add the RF-specific hyperparameters to the `if model is None` statement?

Reply:
the code assumes so far that all constructors can accept all arguments even if they will not use them. I don't want to implement a complicated logic only to prevent that some arguments are not used.